overlord/overlord.go

```diff
@@ -0,0 +1,84 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// Package overlord implements the policies for state transitions for the operation of a snappy system.
+package overlord
+
+// Overlord is the central manager of a snappy system, keeping
+// track of all available state managers and related helpers.
+type Overlord struct {
+	// managers
+	snapMgr   *SnapManager
+	assertMgr *AssertManager
+	skillMgr  *SkillManager
+}
+
+// New creates a new Overlord with all its state managers.
+func New() (*Overlord, error) {
+	stateEng := NewStateEngine()
```

Comment:
yes, we need to pass it when we make the StateJournal, something I have postponed